### PR TITLE
Viper platform fixes

### PR DESCRIPTION
**Pull request by raveenp**
This PR contains following Viper platform fixes:
- soc: viper: update registers and interrupt numbers
- soc: arm: bcm_vk: viper: Increase default NUM_IRQS

### CODE REVIEW

**Review by galak — COMMENTED** [review]
It would probably be best to split this into at least 3 PRs:
* dma changes / fixes
* iproc changes / fixes
* platform changes / fixes

File: boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig [hunk 1]
```diff
@@ -25,3 +25,9 @@ CONFIG_UART_CONSOLE=y
 CONFIG_PCIE_ENDPOINT=y
 CONFIG_PCIE_EP_IPROC=y
 CONFIG_PCIE_EP_IPROC_V2=y
+
+# Enable PAXDMA
+CONFIG_DMA=y
+CONFIG_DMA_64BIT=y
```
Comment (galak) [inline boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig:31]: What about DMA_PL330?

File: boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig [hunk 2]
```diff
@@ -18,3 +18,13 @@ CONFIG_UART_NS16550=y
...
 CONFIG_UART_CONSOLE=y
+
+# Enable PCIe EP
+CONFIG_PCIE_ENDPOINT=y
+CONFIG_PCIE_EP_IPROC=y
+CONFIG_PCIE_EP_IPROC_V2=y
+
+# Enable PAXDMA
+CONFIG_DMA=y
+CONFIG_DMA_IPROC_PAX_V2=y
```
Comment (galak) [inline boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:29]: What about DMA_PL330?

**Comment by olofj** [comment]
Besides the comments from Kumar, please also provide better descriptions in the commits (or in the Pull request). The "why" is missing -- in particular why basic changes such as remapping of interrupts is needed, or why they were wrong before.

**Comment by raveenp** [comment]
> It would probably be best to split this into at least 3 PRs:
> 
> * dma changes / fixes
> * iproc changes / fixes
> * platform changes / fixes

Sure, I will split the patch and push in separate PR's.

File: boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig (hunk 1 above)
Comment (raveenp) [reply boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig:31]: Will enable in https://github.com/zephyrproject-rtos/zephyr/pull/32316

File: boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig (hunk 2 above)
Comment (raveenp) [reply boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:29]: Will enable in https://github.com/zephyrproject-rtos/zephyr/pull/32316

**Comment by raveenp** [comment]
> Besides the comments from Kumar, please also provide better descriptions in the commits (or in the Pull request). The "why" is missing -- in particular why basic changes such as remapping of interrupts is needed, or why they were wrong before.

Sure, will provide the descriptions in the commit.

**Review by galak — APPROVED** [review]

**Review by Abhishek-brcm — APPROVED** [review]

**Pull request merged**